app/v2/notifications/post_notifications.py

```diff
@@ -98,6 +113,49 @@ def post_precompiled_letter_notification():
     return jsonify(resp), 201
 
 
+@v2_notification_blueprint.route("/bulk", methods=["POST"])
+def post_bulk():
+    try:
+        request_json = request.get_json()
+    except werkzeug.exceptions.BadRequest as e:
+        raise BadRequestError(message=f"Error decoding arguments: {e.description}", status_code=400)
+
+    max_rows = current_app.config["CSV_MAX_ROWS"]
+    form = validate(request_json, post_bulk_request(max_rows))
+
+    if len([source for source in [form.get("rows"), form.get("csv")] if source]) != 1:
+        raise BadRequestError(message="You should specify either rows or csv", status_code=400)
+    template = validate_template_exists(form["template_id"], authenticated_service)
+    check_service_has_permission(template.template_type, authenticated_service.permissions)
+
+    remaining_messages = authenticated_service.message_limit - fetch_todays_total_message_count(authenticated_service.id)
+
+    try:
+        if form.get("rows"):
+            output = StringIO()
+            writer = csv.writer(output)
+            writer.writerows(form["rows"])
+            file_data = output.getvalue()
+        else:
+            file_data = form["csv"]
+
+        recipient_csv = RecipientCSV(
```

Comment:
Most of the magic comes from https://github.com/cds-snc/notification-utils/blob/59645a9aed395ffa08f28d567bdb8e73c9bcdfca/notifications_utils/recipients.py#L49